### ISSUE TITLE
Power of parameters

### ISSUE THREAD

**Issue by duguyipiao** [cvxgrp/cvxpy#1044]
**Describe the bug**
I updated to cvxpy 1.1.0. The power of parameters gives something odd.

**To Reproduce**
```python
import numpy as np
import cvxpy as cvx

s = cvx.Parameter(1, nonneg=True)
x = cvx.Variable(1)

obj = cvx.Maximize(x+s)
cons = [x <= 1]
prob = cvx.Problem(obj, cons)

s.value = np.array([1.])
prob.solve(solver='SCS')
```
This gives 2, which is normal.

```python
import numpy as np
import cvxpy as cvx

s = cvx.Parameter(1, nonneg=True)
x = cvx.Variable(1)

obj = cvx.Maximize(x+s**2)
cons = [x <= 1]
prob = cvx.Problem(obj, cons)

s.value = np.array([1.])
prob.solve(solver='SCS')
```
This gives `inf`, which is obviously wrong.

```python
import numpy as np
import cvxpy as cvx

s = cvx.Parameter(1, nonneg=True)
x = cvx.Variable(1)

obj = cvx.Maximize(x*s**2)
cons = [x <= 1]
prob = cvx.Problem(obj, cons)

s.value = np.array([1.])
prob.solve(solver='SCS')
```
This throws an `AssertionError`.

This also happens to the vectors, e.g.
```python
import numpy as np
import cvxpy as cvx

s = cvx.Parameter(2, nonneg=True)
x = cvx.Variable(1)

obj = cvx.Maximize(x+cvx.sum(s**2))
cons = [x <= 1]
prob = cvx.Problem(obj, cons)

s.value = np.array([1.,2.])
prob.solve(solver='SCS')
```
also gives `inf`.

**Version**
 - OS: Linux
 - CVXPY Version: 1.1.0

**Comment by akshayka** [comment]
Thanks for reporting! We'll fix this soon ...

**Comment by akshayka** [comment]
The fix has been merged into master. I'll close this issue once the fix is on PyPI.

**Comment by akshayka** [comment]
The fix is now available on PyPI, version 1.1.1.

Thank you for reporting this issue (and thanks for providing the code snippets that reproduced it).

**Comment by duguyipiao** [comment]
Thanks for the fix. The latest version works for me now.
